src/Plugins/Nop.Plugin.Seed.ProductSync/Clients/InfigoClient.cs

```diff
@@ -8,80 +8,75 @@
 
 namespace Nop.Plugin.Seed.ProductSync.Clients;
 
-public class InfigoClient:IInfigoClient
+public class InfigoClient : IInfigoClient
 {
-    
     private readonly ProductSyncSettings _settings;
     private readonly RestClient _client;
 
     public InfigoClient(ProductSyncSettings settings)
     {
         _settings = settings;
-        if(IsConfigured(_settings))
+        EnsureConfigured();
```

Comment:
are you sure you want to do this here? otherwise this will throw exception in DI if not configured, give a try